src/dsp_tools/commands/xmlupload/check_consistency_with_ontology.py

```diff
@@ -155,3 +162,85 @@ def _get_separate_prefix_and_iri_from_onto_prop_or_cls(
         return tuple(prop_or_cls.split(":"))
     else:
         return None, None
+
+
+def _analyse_all_text_value_encodings_are_correct(
+    root: etree._Element, text_prop_look_up: PropertyTextValueEncodingTypes
+) -> None:
+    """
+    This function analyses if all the encodings for the <text> elements are correct
+    with respect to the specification in the ontology.
+
+    For example, in the ontology, it says that `:hasSimpleText` is without mark-up.
+    The encoding has to be `utf8`
+
+
+    This is correct:
+    ```
+    <text-prop name=":hasSimpleText">
+        <text permissions="prop-default" encoding="utf8">Dies ist ein einfacher Text ohne Markup</text>
+    </text-prop>
+    ```
+
+    This is wrong:
+    ```
+    <text-prop name=":hasSimpleText">
+        <text permissions="prop-default" encoding="xml">Dies ist ein einfacher Text ohne Markup</text>
+    </text-prop>
+    ```
+
+    The accepted encodings are `xml` or `utf8`
+
+    Args:
+        root: root of the data xml document
+        text_prop_look_up: a lookup containing the property names and their specified types
+
+    Raises:
+        InputError: if any <text> elements have a wrong encoding.
+    """
+    text_value_in_data = _get_all_ids_prop_encoding_from_root(root)
+    all_checked = [x for x in text_value_in_data if not _check_correctness_one_prop(x, text_prop_look_up)]
+    if len(all_checked) != 0:
+        msg, df = InvalidTextValueEncodings(all_checked).execute_problem_protocol()
+        if df:
+            csv_file = f"text_value_encoding_errors_{datetime.now().strftime('%Y-%m-%d_%H%M%S')}.csv"
+            csv_path = Path(Path.cwd(), csv_file)
+            df.to_csv(path_or_buf=csv_path, index=False)
+            msg += (
+                "\n\n---------------------------------------\n\n"
+                f"All the problems are listed in the file: '{csv_path.absolute()}'"
+            )
+        raise InputError(msg)
+
+
+def _get_all_ids_prop_encoding_from_root(root: etree._Element) -> list[TextValueData]:
+    res_list = []
+    for res_input in root.iterchildren(tag="resource"):
+        res_list.extend(_get_id_prop_encoding_from_one_resource(res_input))
+    return res_list
+
+
+def _get_id_prop_encoding_from_one_resource(resource: etree._Element) -> list[TextValueData]:
+    res_id = resource.attrib["id"]
+    return [_get_prop_encoding_from_one_property(res_id, child) for child in resource.iterchildren(tag="text-prop")]
+
+
+def _get_prop_encoding_from_one_property(res_id: str, property: etree._Element) -> TextValueData:
+    prop_name = property.attrib["name"]
+    encoding = cast(AllowedEncodings, next(x.attrib["encoding"] for x in property.iterchildren()))
```

Comment:
The only way I managed to make mypy happy was by casting it. Otherwise I got the following error:
```
error: Argument 3 to "TextValueData" has incompatible type "str"; expected "Literal['utf8', 'xml']"  [arg-type]
```
If you know a better way to fix this I would be happy for suggestions.